data/moves.js

```diff
@@ -6977,9 +6983,11 @@ exports.BattleMovedex = {
 		pp: 30,
 		priority: 0,
 		flags: {},
-		onTryHit: function (target, source) {
-			this.add('-activate', target, 'move: Happy Hour');
-			return null;
+		sideCondition: 'happyhour',
+		effect: {
+			onStart: function (side) {
+				this.add('-activate', side, 'move: Happy Hour');
+			},
```

Comment:
There's no need to "implement" Happy Hour, since it never fails. You can use it every turn without failure, like Splash.

Reply:
Huh, was remembering that wrong.

Reply:
Regardless, Happy Hour should have a message.